train.py

```diff
@@ -37,51 +39,74 @@
+    batch_ph = tf.placeholder(tf.int32, [None, SEQUENCE_LENGTH], name='batch_ph')
+    target_ph = tf.placeholder(tf.float32, [None], name='target_ph')
+    seq_len_ph = tf.placeholder(tf.int32, [None], name='seq_len_ph')
+    keep_prob_ph = tf.placeholder(tf.float32, name='keep_prob_ph')
 
 # Embedding layer
-embeddings_var = tf.Variable(tf.random_uniform([vocabulary_size, EMBEDDING_DIM], -1.0, 1.0), trainable=True)
-batch_embedded = tf.nn.embedding_lookup(embeddings_var, batch_ph)
+with tf.name_scope('Embedding_layer'):
+    embeddings_var = tf.Variable(tf.random_uniform([vocabulary_size, EMBEDDING_DIM], -1.0, 1.0), trainable=True)
+    tf.summary.histogram('embeddings_var', embeddings_var)   
+    batch_embedded = tf.nn.embedding_lookup(embeddings_var, batch_ph)
 
 # (Bi-)RNN layer(-s)
 rnn_outputs, _ = bi_rnn(GRUCell(HIDDEN_SIZE), GRUCell(HIDDEN_SIZE),
                         inputs=batch_embedded, sequence_length=seq_len_ph, dtype=tf.float32)
+tf.summary.histogram('rnn_outputs', rnn_outputs)   
 # rnn_outputs, _ = rnn(GRUCell(hidden_size), inputs=batch_embedded, sequence_length=seq_len_ph, dtype=tf.float32)
 
+
 # Attention layer
-attention_output, alphas = attention(rnn_outputs, ATTENTION_SIZE, return_alphas=True)
+with tf.name_scope('Attention_layer'):
+    attention_output, alphas = attention(rnn_outputs, ATTENTION_SIZE, return_alphas=True)
+    tf.summary.histogram('alphas', alphas)   
 
 # Dropout
 drop = tf.nn.dropout(attention_output, keep_prob_ph)
-
+    
 # Fully connected layer
-W = tf.Variable(tf.truncated_normal([HIDDEN_SIZE * 2, 1], stddev=0.1))  # Hidden size is multiplied by 2 for Bi-RNN
-b = tf.Variable(tf.constant(0., shape=[1]))
-y_hat = tf.nn.xw_plus_b(drop, W, b)
-y_hat = tf.squeeze(y_hat)
-
-# Cross-entropy loss and optimizer initialization
-loss = tf.reduce_mean(tf.nn.sigmoid_cross_entropy_with_logits(logits=y_hat, labels=target_ph))
-optimizer = tf.train.AdamOptimizer(learning_rate=1e-3).minimize(loss)
-
-# Accuracy metric
-accuracy = tf.reduce_mean(tf.cast(tf.equal(tf.round(tf.sigmoid(y_hat)), target_ph), tf.float32))
+with tf.name_scope('Fully_connected_layer'):
+    W = tf.Variable(tf.truncated_normal([HIDDEN_SIZE * 2, 1], stddev=0.1))  # Hidden size is multiplied by 2 for Bi-RNN
+    b = tf.Variable(tf.constant(0., shape=[1]))
+    y_hat = tf.nn.xw_plus_b(drop, W, b)
+    y_hat = tf.squeeze(y_hat)
+    tf.summary.histogram('W', W)   
+
+with tf.name_scope('Metrics'):
+    # Cross-entropy loss and optimizer initialization
+    loss = tf.reduce_mean(tf.nn.sigmoid_cross_entropy_with_logits(logits=y_hat, labels=target_ph))
+    tf.summary.scalar('loss', loss)
+    optimizer = tf.train.AdamOptimizer(learning_rate=1e-3).minimize(loss)
+    
+    # Accuracy metric
+    accuracy = tf.reduce_mean(tf.cast(tf.equal(tf.round(tf.sigmoid(y_hat)), target_ph), tf.float32))
+    tf.summary.scalar('accuracy', accuracy)
 
 # Actual lengths of sequences
 seq_len_test = np.array([list(x).index(0) + 1 for x in X_test])
```

Comment:
These are unused variables, just noticed. Remove them, please

Reply:
I don't quite understand, did you means removing the variable 'accuracy'? I guess accuracy is necessary in training progress, especially when someone decides to disable tensorboard logging, he can still get an accuracy on training progress.

Reply:
OK. I cleaned them, and I don't find any other unused variables there.